From 4e2018f8a4daa76ca87b3980c248e7e9fc25242d Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Anders=20F=20Bj=C3=B6rklund?= <anders.f.bjorklund@gmail.com>
Date: Sat, 17 Mar 2018 15:19:09 +0100
Subject: [PATCH] Use mem_sreclaimable metric, instead of mem_slab

---
 graph.d/mem_report.php | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/graph.d/mem_report.php b/graph.d/mem_report.php
index fc18d1c0..d443965a 100644
--- a/graph.d/mem_report.php
+++ b/graph.d/mem_report.php
@@ -55,9 +55,9 @@ function graph_mem_report ( &$rrdtool_graph ) {
            ."CDEF:'bmem_shared'=mem_shared,1024,* ";
     }
 
-    if (file_exists("$rrd_dir/mem_slab.rrd")) {
+    if (file_exists("$rrd_dir/mem_sreclaimable.rrd")) {
        $bmem_used_cdef .= ",'bmem_slab',UN,0,'bmem_slab',IF,-";
-       $bmem_slab_defs = "DEF:'mem_slab'='${rrd_dir}/mem_slab.rrd':'sum':AVERAGE "
+       $bmem_slab_defs = "DEF:'mem_slab'='${rrd_dir}/mem_sreclaimable.rrd':'sum':AVERAGE "
            ."CDEF:'bmem_slab'=mem_slab,1024,* ";
     }
 
@@ -121,7 +121,7 @@ function graph_mem_report ( &$rrdtool_graph ) {
                 . "GPRINT:'cached_max':'${space1}Max\:%6.1lf%s\\l' ";
     }
 
-    if (file_exists("$rrd_dir/mem_slab.rrd")) {
+    if (file_exists("$rrd_dir/mem_sreclaimable.rrd")) {
         $series .= "STACK:'bmem_slab'#${conf['mem_slab_color']}:'Slab${rmspace}' ";
 
         if ( $conf['graphreport_stats'] ) {
