From be1ae9a2d5c496861dad5806d06470935e8374f8 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Wed, 3 May 2017 13:22:43 -0400
Subject: [PATCH] Fix lib suffix

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 1de1202..bdf3e96 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -54,7 +54,7 @@ include_directories(${OPENSSL_INCLUDE_DIR} ${ZLIB_INCLUDE_DIR} ${EXPAT_INCLUDE_D
 if(NOT APPLE AND NOT WIN32)
 set_source_files_properties(location.cpp PROPERTIES COMPILE_DEFINITIONS CONFDIR="${CMAKE_INSTALL_PREFIX}/share/TrustedQSL/")
 set(HEADERS_TO_INSTALL tqsllib.h tqslerrno.h cabrillo.h adif.h tqslconvert.h)
-install(TARGETS tqsllib DESTINATION lib$(LIB_SUFFIX))
+install(TARGETS tqsllib DESTINATION lib${LIB_SUFFIX})
 install(FILES config.xml DESTINATION share/TrustedQSL)
 install(FILES ${HEADERS_TO_INSTALL} DESTINATION include)
 endif()
-- 
Daniel M. Weeks

