diff -r 0f9d3345c850 pqlop.py
--- a/pqlop.py	Thu Jan 26 00:43:24 2012 +0400
+++ b/pqlop.py	Mon Jul 25 20:48:18 2016 -0400
@@ -81,7 +81,7 @@
     timesecs //= 365
     years = timesecs
     if years:
-        print("%s %s," % (colorize(GREEN, days), "year"+setmod(years)), end=' ')
+        print("%s %s," % (colorize(GREEN, years), "year"+setmod(years)), end=' ')
     if days:
         print("%s %s," % (colorize(GREEN, days), "day"+setmod(days)), end=' ')
     if hours:
