From 4371c5440d4d9b84a21373018d7f6c4e40171775 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Thu, 20 Oct 2016 23:38:39 -0400
Subject: [PATCH] Correct type of deadYet

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 server/faker.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/server/faker.h b/server/faker.h
index 3c8b648..90e88fd 100644
--- a/server/faker.h
+++ b/server/faker.h
@@ -30,7 +30,7 @@ namespace vglfaker
 {
 	extern Display *dpy3D;
 	extern void safeExit(int);
-	extern int deadYet;
+	extern bool deadYet;
 	extern long getTraceLevel(void);
 	extern void setTraceLevel(long level);
 	#ifdef FAKEXCB
-- 
Daniel M. Weeks

