diff --git a/Makefile.orig b/Makefile
index e3eb645..ff26b51 100644
--- a/Makefile.orig
+++ b/Makefile
@@ -17,12 +17,12 @@ BUILD_DIR = out
 
 # Installation directories by convention
 # http://www.gnu.org/prep/standards/html_node/Directory-Variables.html
-PREFIX = /usr/local
-EXEC_PREFIX = $(PREFIX)
-BINDIR = $(EXEC_PREFIX)/bin
-DATAROOTDIR = $(PREFIX)/share
+DESTDIR = /usr/local
+EXEC_DESTDIR = $(DESTDIR)
+BINDIR = $(EXEC_DESTDIR)/bin
+DATAROOTDIR = $(DESTDIR)/share
 DOCDIR = $(DATAROOTDIR)/doc/$(CPROG)
-SYSCONFDIR = $(PREFIX)/etc
+SYSCONFDIR = $(DESTDIR)/etc
 HTMLDIR = $(DOCDIR)
 
 UNAME := $(shell uname)
@@ -30,7 +30,7 @@ UNAME := $(shell uname)
 # desired configuration of the document root
 # never assume that the document_root actually
 # exists on the build machine.  When building
-# a chroot, PREFIX if just a directory which
+# a chroot, DESTDIR if just a directory which
 # later becomes /.
 DOCUMENT_ROOT = $(HTMLDIR)
 PORTS = 8080
@@ -144,7 +144,7 @@ help:
 	@echo "   PORTS=8080            listening ports override when installing"
 	@echo "   SSL_LIB=libssl.so.0   use versioned SSL library"
 	@echo "   CRYPTO_LIB=libcrypto.so.0 system versioned CRYPTO library"
-	@echo "   PREFIX=/usr/local     sets the install directory"
+	@echo "   DESTDIR=/usr/local     sets the install directory"
 	@echo "   COPT='-DNO_SSL'       method to insert compile flags"
 	@echo ""
 	@echo " Compile Flags"
