From 3051292c4502f9d52dc9a93335dc786afc707214 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Mon, 20 Mar 2017 16:50:06 -0400
Subject: [PATCH 2/4] flags

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 config.mk        | 4 ++--
 iceprog/Makefile | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/config.mk b/config.mk
index 7b3b494..2d45013 100644
--- a/config.mk
+++ b/config.mk
@@ -16,8 +16,8 @@ endif
 WARN_LEVEL ?= all
 
 LDLIBS = -lm -lstdc++
-CFLAGS += -MD -O$(OPT_LEVEL) $(DBG_LEVEL) -W$(WARN_LEVEL) -std=$(C_STD) -I$(PREFIX)/include
-CXXFLAGS += -MD -O$(OPT_LEVEL) $(DBG_LEVEL) -W$(WARN_LEVEL) -std=$(CXX_STD) -I$(PREFIX)/include
+CFLAGS += -MD $(DBG_LEVEL) -W$(WARN_LEVEL) -std=$(C_STD) -I$(PREFIX)/include
+CXXFLAGS += -MD $(DBG_LEVEL) -W$(WARN_LEVEL) -std=$(CXX_STD) -I$(PREFIX)/include
 
 DESTDIR ?=
 CHIPDB_SUBDIR ?= icebox
diff --git a/iceprog/Makefile b/iceprog/Makefile
index 8739ba6..03b4b8b 100644
--- a/iceprog/Makefile
+++ b/iceprog/Makefile
@@ -4,7 +4,7 @@ ifneq ($(shell uname -s),Darwin)
   LDLIBS = -L/usr/local/lib -lm
 else
   LIBFTDI_NAME = $(shell $(PKG_CONFIG) --exists libftdi1 && echo ftdi1 || echo ftdi)
-  LDLIBS = -L/usr/local/lib -l$(LIBFTDI_NAME) -lm
+  LDLIBS = -l$(LIBFTDI_NAME) -lm
 endif
 
 ifeq ($(STATIC),1)
