From 4cb66849a3e9933dbdebf40ebfdea6437fd03e58 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Mon, 20 Mar 2017 16:51:37 -0400
Subject: [PATCH 3/4] No iceprog

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 518e08c..ec7d623 100644
--- a/Makefile
+++ b/Makefile
@@ -1,4 +1,4 @@
-SUBDIRS = icebox icepack iceprog icemulti icepll icetime icebram
+SUBDIRS = icebox icepack icemulti icepll icetime icebram
 
 include config.mk
 
