From 867470f8ed261c54f810ea4e73fe63e9ce302b62 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Mon, 20 Mar 2017 16:51:37 -0400
Subject: [PATCH 3/3] No iceprog

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 79e7a7f..2c22b3b 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 include config.mk
 
-SUBDIRS = icebox icepack iceprog icemulti icepll icetime icebram
+SUBDIRS = icebox icepack icemulti icepll icetime icebram
 
 all clean install uninstall:
 	for dir in $(SUBDIRS); do \
