From b6580c62acd4f5720532b5f1d7de4459be2821d7 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Mon, 20 Mar 2017 16:50:06 -0400
Subject: [PATCH 2/4] flags

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 config.mk        | 4 ++--
 iceprog/Makefile | 2 +-
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/config.mk b/config.mk
index 364f8bc..7f0455e 100644
--- a/config.mk
+++ b/config.mk
@@ -1,8 +1,8 @@
 CXX ?= clang++
 CC ?= clang
 LDLIBS = -lm -lstdc++
-CFLAGS += -MD -O0 -ggdb -Wall -std=c99 -I/usr/local/include
-CXXFLAGS += -MD -O0 -ggdb -Wall -std=c++11 -I/usr/local/include
+override CFLAGS += -std=c99
+override CXXFLAGS += -std=c++11
 PKG_CONFIG ?= pkg-config
 DESTDIR ?=
 PREFIX ?= /usr/local
diff --git a/iceprog/Makefile b/iceprog/Makefile
index d71a9b7..6d32a93 100644
--- a/iceprog/Makefile
+++ b/iceprog/Makefile
@@ -4,7 +4,7 @@ ifneq ($(shell uname -s),Darwin)
   LDLIBS = -L/usr/local/lib -lm
 else
   LIBFTDI_NAME = $(shell $(PKG_CONFIG) --exists libftdi1 && echo ftdi1 || echo ftdi)
-  LDLIBS = -L/usr/local/lib -l$(LIBFTDI_NAME) -lm
+  LDLIBS = -l$(LIBFTDI_NAME) -lm
 endif
 
 ifeq ($(STATIC),1)
