From 0534be0b6c23b67e0d4ce874dba85e8f58b1e7df Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Fri, 6 Oct 2017 13:18:53 -0400
Subject: [PATCH] Use zmq shared library

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 092e8299..b024827a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -424,7 +424,7 @@ if(USE_DIST_KVSTORE)
     target_link_libraries(mxnet PUBLIC optimized ${pslite_LINKER_LIBS_RELEASE})
 
   else()
-    set(pslite_LINKER_LIBS protobuf zmq-static)
+    set(pslite_LINKER_LIBS protobuf zmq)
   endif()
   add_definitions(-DMXNET_USE_DIST_KVSTORE)
   include_directories(SYSTEM ${pslite_INCLUDE_DIR})
-- 
Daniel M. Weeks

