From 0bc5693772cc4405067ddd53e5aaea6ca1dfaca7 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Thu, 23 Mar 2017 11:57:24 -0400
Subject: [PATCH] OpWrapperGenerator not Py3 compatible

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 204831e6..6d7dd7ad 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -488,7 +488,7 @@ if(NOT MSVC)
   add_custom_command(TARGET mxnet
     POST_BUILD
     COMMAND cp $<TARGET_FILE:mxnet> .
-    COMMAND python OpWrapperGenerator.py $<TARGET_FILE_NAME:mxnet>
+    COMMAND python2 OpWrapperGenerator.py $<TARGET_FILE_NAME:mxnet>
     COMMAND rm $<TARGET_FILE_NAME:mxnet>
     WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}/cpp-package/src/OpWrapperGenerator/
   )
