From 171428f1d0309fd242065171cf4a98cc710d2e84 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Fri, 6 Oct 2017 15:39:07 -0400
Subject: [PATCH 2/2] Use shared pslite

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b024827a..fc989b7b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -424,7 +424,7 @@ if(USE_DIST_KVSTORE)
     target_link_libraries(mxnet PUBLIC optimized ${pslite_LINKER_LIBS_RELEASE})
 
   else()
-    set(pslite_LINKER_LIBS protobuf zmq)
+    set(pslite_LINKER_LIBS protobuf zmq pslite)
   endif()
   add_definitions(-DMXNET_USE_DIST_KVSTORE)
   include_directories(SYSTEM ${pslite_INCLUDE_DIR})
-- 
Daniel M. Weeks

