From 6f175149448d2cf826e517e614a81c8b19403697 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Thu, 2 Aug 2018 01:13:29 -0400
Subject: [PATCH] Use shared MKLDNN

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 8c3e63568..68538502b 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -217,16 +217,6 @@ if(USE_TENSORRT)
 endif()
 
 if(USE_MKLDNN)
-  include(cmake/MklDnn.cmake)
-  # CPU architecture (e.g., C5) can't run on another architecture (e.g., g3).
-  if(NOT MSVC)
-    set(ARCH_OPT_FLAGS "-mtune=generic")
-  endif()
-  set(WITH_TEST OFF)
-  set(WITH_EXAMPLE OFF)
-  add_subdirectory(3rdparty/mkldnn)
-
-  include_directories(3rdparty/mkldnn/include)
   add_definitions(-DUSE_MKL=1)
   add_definitions(-DCUB_MKL=1)
   add_definitions(-DMXNET_USE_MKLDNN=1)
-- 
Daniel M. Weeks

