From 2224e92136882f98f2d7f950dbcaf8e5a07dac0f Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Tue, 20 Mar 2018 12:56:13 -0400
Subject: [PATCH 1/2] Use shared gtest

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 9 +--------
 1 file changed, 1 insertion(+), 8 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b3a89558..3e03eac2 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -369,14 +369,7 @@ if(USE_JEMALLOC)
 endif()
 
 include(CTest)
-set(GTEST_ROOT "${CMAKE_CURRENT_SOURCE_DIR}/3rdparty/googletest/googletest")
-set(GTEST_INCLUDE_DIR ${GTEST_ROOT}/include)
-#set(GTEST_BOTH_LIBRARIES gtest gtest_main)
-set(GTEST_LIBRARIES gtest gtest_main)
-set(GTEST_MAIN_LIBRARY gtest_main)
-set(GTEST_LIBRARY gtest)
-
-add_subdirectory(${GTEST_ROOT})
+
 find_package(GTest REQUIRED)
 
 # cudnn detection
-- 
Daniel M. Weeks

