diff --git a/setup.py b/setup.py
index 3e395f5..7228a52 100644
--- a/setup.py
+++ b/setup.py
@@ -54,22 +54,6 @@ else:
   from distutils.core import setup
   setup_kwargs = dict()
 
-
-storage_dirs = [ ('storage/ceres', []), ('storage/whisper',[]),
-                 ('storage/lists',[]), ('storage/log',[]),
-                 ('storage/rrd',[]) ]
-conf_files = [ ('conf', glob('conf/*.example')) ]
-
-install_files = storage_dirs + conf_files
-
-# Let's include redhat init scripts, despite build platform
-# but won't put them in /etc/init.d/ automatically anymore
-init_scripts = [ ('examples/init.d', ['distro/redhat/init.d/carbon-cache',
-                                      'distro/redhat/init.d/carbon-relay',
-                                      'distro/redhat/init.d/carbon-aggregator']) ]
-install_files += init_scripts
-
-
 try:
     setup(
         name='carbon',
@@ -84,7 +68,6 @@ try:
         package_dir={'' : 'lib'},
         scripts=glob('bin/*'),
         package_data={ 'carbon' : ['*.xml'] },
-        data_files=install_files,
         install_requires=['Twisted', 'txAMQP', 'cachetools'],
         classifiers=(
             'Intended Audience :: Developers',
