diff --git a/src/channel.c b/src/channel.c
index 3a5fac0..4cf1482 100644
--- a/src/channel.c
+++ b/src/channel.c
@@ -285,6 +285,7 @@ int Chan_userLeave(client_t *client)
 	if (client->channel) {
 		list_del(&client->chan_node);
 		leaving = (channel_t *)client->channel;
+		leaving->clientcount--;
 		if (leaving->temporary && list_empty(&leaving->clients)) {
 			leaving_id = leaving->id;
 			Chan_freeChannel(leaving);
@@ -306,6 +307,7 @@ int Chan_userJoin(channel_t *ch, client_t *client)
 	leaving_id = Chan_userLeave(client);
 	list_add_tail(&client->chan_node, &ch->clients);
 	client->channel = (void *)ch;
+	ch->clientcount++;
 	return leaving_id;
 }
 
diff --git a/src/channel.h b/src/channel.h
index d348184..7898132 100644
--- a/src/channel.h
+++ b/src/channel.h
@@ -45,6 +45,7 @@ typedef struct channel {
 	int position;
 	struct dlist node;
 	struct dlist subs;
+	int clientcount;
 	struct dlist clients;
 	struct dlist flatlist_node;
 	struct dlist channel_links;
