--- CMakeLists.txt.orig	2015-03-26 10:47:17.549901938 -0400
+++ CMakeLists.txt	2015-03-26 10:56:24.470901696 -0400
@@ -2,6 +2,9 @@
 
 cmake_minimum_required(VERSION 2.6)
 
+option(BUILD_DOC "Build documentation" ON)
+option(BUILD_TESTS "Build tests" ON)
+
 # Set the default CMAKE_BUILD_TYPE to Release.
 # This should be done before the project command since the latter can set
 # CMAKE_BUILD_TYPE itself (it does so for nmake).
@@ -82,8 +85,11 @@
   add_library(testformat STATIC ${FMT_SOURCE_FILES} ${src})
 endif ()
 
-add_subdirectory(doc)
+if (BUILD_DOC)
+  add_subdirectory(doc)
+endif (BUILD_DOC)
 
+if (BUILD_TESTS)
 include_directories(. gmock)
 
 # We compile Google Test ourselves instead of using pre-compiled libraries.
@@ -168,6 +174,7 @@
   set(CPACK_RESOURCE_FILE_README ${FORMAT_SOURCE_DIR}/README.rst)
   include(CPack)
 endif ()
+endif (BUILD_TESTS)
 
 # Install our targets
 install(TARGETS format DESTINATION lib)
