diff --git a/CMakeLists.txt b/CMakeLists.txt
new file mode 100644
index 0000000..386bdaf
--- /dev/null
+++ CMakeLists.txt
@@ -0,0 +1,6 @@
+cmake_minimum_required(VERSION 2.8.9)
+
+SET(BUILD_SHARED_LIBS ON)
+
+add_subdirectory(Detour)
+add_subdirectory(Recast)
diff --git a/Detour/CMakeLists.txt b/Detour/CMakeLists.txt
new file mode 100644
index 0000000..7737743
--- /dev/null
+++ Detour/CMakeLists.txt
@@ -0,0 +1,27 @@
+set(Detour_SRCS
+    Source/DetourAlloc.cpp
+    Source/DetourCommon.cpp
+    Source/DetourNavMesh.cpp
+    Source/DetourNavMeshBuilder.cpp
+    Source/DetourNavMeshQuery.cpp
+    Source/DetourNode.cpp
+)
+
+set(Detour_HEADERS
+    Include/DetourAlloc.h
+    Include/DetourAssert.h
+    Include/DetourCommon.h
+    Include/DetourMath.h
+    Include/DetourNavMeshBuilder.h
+    Include/DetourNavMesh.h
+    Include/DetourNavMeshQuery.h
+    Include/DetourNode.h
+    Include/DetourStatus.h
+)
+
+include_directories(Include)
+
+add_library(Detour ${Detour_SRCS})
+
+install(TARGETS Detour DESTINATION lib${LIB_SUFFIX})
+install(FILES ${Detour_HEADERS} DESTINATION include)
diff --git a/Recast/CMakeLists.txt b/Recast/CMakeLists.txt
new file mode 100644
index 0000000..93d0847
--- /dev/null
+++ Recast/CMakeLists.txt
@@ -0,0 +1,25 @@
+set(Recast_SRCS
+    Source/Recast.cpp
+    Source/RecastAlloc.cpp
+    Source/RecastArea.cpp
+    Source/RecastContour.cpp
+    Source/RecastFilter.cpp
+    Source/RecastLayers.cpp
+    Source/RecastMesh.cpp
+    Source/RecastMeshDetail.cpp
+    Source/RecastRasterization.cpp
+    Source/RecastRegion.cpp
+)
+
+set(Recast_HEADERS
+    Include/RecastAlloc.h
+    Include/RecastAssert.h
+    Include/Recast.h
+)
+
+include_directories(Include)
+
+add_library(Recast ${Recast_SRCS})
+
+install(TARGETS Recast DESTINATION lib${LIB_SUFFIX})
+install(FILES ${Recast_HEADERS} DESTINATION include)
