From abf834a5486794a6d5d962978c0b7aee890b8d3b Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Wed, 31 May 2017 09:40:49 -0400
Subject: [PATCH 3/3] No doc

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 1 -
 1 file changed, 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d2f94eb..281f573 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -132,7 +132,6 @@ endif()
 # ---[ Install the archive static lib and header files
 install(TARGETS dmlc DESTINATION lib${LIB_SUFFIX})
 install(DIRECTORY include DESTINATION .)
-install(DIRECTORY doc DESTINATION .)
 
 # ---[ Linter target
 if(MSVC)
