From 137b4df562d4bd5996e2967fee7f5e2589042c45 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Sun, 13 Aug 2017 22:50:39 -0400
Subject: [PATCH 1/3] Installed into correct lib

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0b3b3ec..557965a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -130,7 +130,7 @@ if(INSTALL_INCLUDE_DIR)
 endif()
 
 # ---[ Install the archive static lib and header files
-install(TARGETS dmlc ARCHIVE DESTINATION lib) 
+install(TARGETS dmlc ARCHIVE DESTINATION lib${LIB_SUFFIX})
 install(DIRECTORY include DESTINATION .)
 install(DIRECTORY doc DESTINATION .)
 
