diff --git a/src/CascLib.h b/src/CascLib.h
index d60a681..190bbb1 100644
--- a/src/CascLib.h
+++ b/src/CascLib.h
@@ -227,6 +227,16 @@ HANDLE WINAPI CascFindFirstFile(HANDLE hStorage, const char * szMask, PCASC_FIND
 bool  WINAPI CascFindNextFile(HANDLE hFind, PCASC_FIND_DATA pFindData);
 bool  WINAPI CascFindClose(HANDLE hFind);
 
+//-----------------------------------------------------------------------------
+// GetLastError/SetLastError support for non-Windows platform
+
+#ifndef PLATFORM_WINDOWS
+
+int GetLastError();
+void SetLastError(int nError);
+
+#endif  // PLATFORM_WINDOWS
+
 #ifdef __cplusplus
 }   // extern "C"
 #endif
diff --git a/src/common/Common.h b/src/common/Common.h
index effc8a3..557878b 100644
--- a/src/common/Common.h
+++ b/src/common/Common.h
@@ -29,24 +29,6 @@ extern unsigned char AsciiToUpperTable_BkSlash[256];
 extern unsigned char IntToHexChar[];
 
 //-----------------------------------------------------------------------------
-// GetLastError/SetLastError support for non-Windows platform
-
-#ifndef PLATFORM_WINDOWS
-
-#ifdef __cplusplus
-extern "C" {
-#endif
-
-int GetLastError();
-void SetLastError(int nError);
-
-#ifdef __cplusplus
-}   // extern "C"
-#endif
-
-#endif  // PLATFORM_WINDOWS
-
-//-----------------------------------------------------------------------------
 // String manipulation
 
 void CopyString(char * szTarget, const char * szSource, size_t cchLength);
