From c755be999007125711edbc95050f837ad8ece0e8 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Fri, 16 Mar 2018 02:06:48 -0400
Subject: [PATCH] Use LIB_SUFFIX

---
 src/CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 2393323b..d09fc385 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -55,5 +55,5 @@ set_property(TARGET ${TARGET_NAME} PROPERTY CXX_STANDARD_REQUIRED ON)
 set_property(TARGET ${TARGET_NAME} PROPERTY VERSION "${PROJECT_VERSION}.0")
 set_property(TARGET ${TARGET_NAME} PROPERTY SOVERSION "0")
 
-install(TARGETS ${TARGET_NAME} DESTINATION lib)
+install(TARGETS ${TARGET_NAME} DESTINATION lib${LIB_SUFFIX})
 install(FILES ${HEADERS} DESTINATION include)
-- 
2.16.2

