From 5d9550939f6a113960b9fbbec35e5fbda00e256f Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Thu, 2 Feb 2017 14:36:33 -0500
Subject: [PATCH] CMake linking consistent with Make

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 cmake/ProtoBuf.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/ProtoBuf.cmake b/cmake/ProtoBuf.cmake
index 43149ac..0faf64a 100644
--- a/cmake/ProtoBuf.cmake
+++ b/cmake/ProtoBuf.cmake
@@ -3,7 +3,7 @@
 
 find_package( Protobuf REQUIRED )
 include_directories(SYSTEM ${PROTOBUF_INCLUDE_DIR})
-list(APPEND pslite_LINKER_LIBS ${PROTOBUF_LIBRARIES})
+list(APPEND pslite_LINKER_LIBS ${PROTOBUF_LITE_LIBRARIES})
 
 # As of Ubuntu 14.04 protoc is no longer a part of libprotobuf-dev package
 # and should be installed separately as in: sudo apt-get install protobuf-compiler
-- 
Daniel M. Weeks

