From 5128482370deeb78918fdd20f8efe3326dd81265 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Wed, 2 Aug 2017 14:29:56 -0400
Subject: [PATCH 2/3] Link protobuf lib

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 1 +
 1 file changed, 1 insertion(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 5498c33..fe827dc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -20,6 +20,7 @@ include_directories(pslite ${ZMQ_INCLUDE_DIRS})
 list(APPEND pslite_LINKER_LIBS_L ${ZMQ_LIBRARIES})
 # ---[ Google-protobuf
 include(cmake/ProtoBuf.cmake)
+list(APPEND pslite_LINKER_LIBS_L ${Protobuf_LITE_LIBRARIES})
 
 # generate protobuf sources
 set(proto_gen_folder "${PROJECT_BINARY_DIR}/src")
