From 4d9e7521784a04af2aaac78fcf7d9c2d9dedfea9 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Wed, 2 Aug 2017 14:30:35 -0400
Subject: [PATCH 3/3] Actually link libraries

Signed-off-by: Daniel M. Weeks <dan@danweeks.net>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index fe827dc..46153fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -53,7 +53,7 @@ endif()
 list(APPEND SOURCE ${proto_srcs}) 
 add_library(pslite ${SOURCE}) 
 
-target_link_libraries(pslite ${pslite_LINKER_LIBS})
+target_link_libraries(pslite ${pslite_LINKER_LIBS_L})
 
 list(APPEND pslite_LINKER_LIBS_L_RELEASE ${Protobuf_LIBRARY})
 list(APPEND pslite_LINKER_LIBS_L_DEBUG ${Protobuf_LIBRARY_DEBUG})
