From 28ca0876b6bed2feb5134bb70b9557242b7880e9 Mon Sep 17 00:00:00 2001
From: "Daniel M. Weeks" <dan@danweeks.net>
Date: Mon, 24 Oct 2016 15:38:44 -0400
Subject: [PATCH] Do not install tests

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 8c8dc69..fe407b3 100755
--- a/setup.py
+++ b/setup.py
@@ -34,7 +34,7 @@ setup(name='cornice',
       author_email='services-dev@mozilla.org',
       url='https://github.com/mozilla-services/cornice',
       keywords='web pyramid pylons',
-      packages=find_packages(),
+      packages=find_packages(exclude=("tests",)),
       package_data=package_data,
       include_package_data=True,
       zip_safe=False,
-- 
2.10.1

